scripts/switcher_gen.py

```diff
@@ -0,0 +1,89 @@
+import os
+import git
+import json
+import re
+
+
+# Filter function for building latest versions of each major tag.
+# For example, out of ["v0.8.0", "v0.8.1", "v0.9.2", "v0.9.3"]
+# those would be ["v0.8.1", "v0.9.3"]
+def latest_tags_filter(tag_list: list, start_version: str = "v0.8.0") -> list:
+    regex = re.compile(r"^v\d+\.\d+\.\d+$")
+    tag_list = list(filter(regex.match, tag_list))
+    latest_tags = {}
+    start_version = str(start_version).replace("v", "").split(".")
+    for tag in tag_list:
+        tag = str(tag).replace("v", "").split(".")
+        tag_group = (tag[0], tag[1])
+        # Not building versions lower than the start version (initially, "v0.8.0")
+        if not (int(tag[0]) == int(start_version[0]) and int(tag[1]) < int(start_version[1])):
```

Comment:
If we ever make start_version higher (e.g. 1.0.0), this won't work.
Need to compare the first part too.
This can be done by converting version strings to tuples of int (`if tuple(map(int, tag)) >= tuple(map(int, start_version))`):
https://stackoverflow.com/questions/5292303/how-does-tuple-comparison-work-in-python

Reply:
940222e